### PR TITLE
Expose isPublic and userCanViewSource

### LINKED ISSUE #1216: Show option for setting content public
We need an option for an author to make content public, which would presumably go into Assignment Settings for now.

We have two flags that should be exposed to the UI:
* isPublic: content can be viewed by anyone, even without logging in (currently through the /public endpoint).
* userCanViewSource: those who can view content can also view the DoenetML.  Currently, only implemented in conjunction with isPublic to allow the DoenetML to be viewed in the Guest Editor.

### PR DESCRIPTION
**Pull request by Lyanthropos**
Closes #1216

### CODE REVIEW

**Review by dqnykamp — CHANGES_REQUESTED** [review]
On a course where I am the owner, I am getting 403 errors coming from updateContentFlags.php when trying to change these (and other) settings from the editor.  They work from the course listing, though.  The problem is presumably due to the fact that the editor does not have courseId url parameter set.  In menus or tools that may be called from an editor or a viewer, we need to get courseId from the courseIdAtom, not the url parameters.

File: src/Api/updateContentFlags.php [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    $userId,
+    $courseId
+);
+if (
+    array_key_exists('isPublic', $proviedValues) &&
+    $permissions['canPublishContent'] != '1'
+) {
+    $success = false;
+    $message = 'You need permission to publish content.';
+    http_response_code(403);
```
[the remark this answers is not in the document]
Comment (Lyanthropos) [reply src/Api/updateContentFlags.php:65]: I'll use the success response until we discuss it further.

File: src/Api/updateContentFlags.php [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+    $result = $conn->query($sql);
+
+    if ($result == false) {
+        $success = false;
+        $message = 'Database error';
+        http_response_code(500);
+    } else {
+        // set response code - 200 OK
+        http_response_code(202);
```
[the remark this answers is not in the document]
Comment (Lyanthropos) [reply src/Api/updateContentFlags.php]: Good to know

**Review by dqnykamp — APPROVED** [review]
looks good

**Pull request merged**